src/files/rsql-filter-list.ts

```diff
@@ -10,11 +10,15 @@ export class RSQLFilterList {
   }
 
   public and(filter: RSQLFilterExpression | RSQLFilterList) {
-    this.andList.push(filter);
+    if (filter !== undefined && filter !== null && filter.build() !== '') {
+      this.andList.push(filter);
+    }
   }
 
   public or(filter: RSQLFilterExpression | RSQLFilterList) {
-    this.orList.push(filter);
+    if (filter !== undefined && filter !== null && filter.build() !== '') {
```

Comment:
It might be worth it to pull this logic into a helper method (e.g. `if (filterPopulated(filter)) {).

Reply:
That is a good idea.  I was thinking about making a `size()` method, but since the lists are either RSQLFilterExpression or RSQLFilterList, it made it difficult to figure out correctly.